pyqtorch/matrices_sparse.py

```diff
@@ -106,40 +119,47 @@ def generate_ising_from_graph(graph,
                 key = (edge[1], edge[0])
             H += precomputed_zz[N][key]
         else:
-            if type_ising == 'Z':
+            if type_ising == "Z":
                 H += ZZ(N, edge[0], edge[1], device).copy()
-            elif type_ising == 'N':
+            elif type_ising == "N":
                 H += NN(N, edge[0], edge[1], device).copy()
             else:
                 raise ValueError("'type_ising' must be in ['Z', 'N']")
 
     return H
 
 
-def general_hamiltonian(graph=None, alpha=None, beta=None, gamma=None, device='cpu'):
-    #alpha, beta, gamma: matrices of parameters (e.g. alpha_ij)
-    #connectivity_graph: which qubits are connected
-    #no 1 body terms
+@typing.no_type_check
```

Comment:
@awennersteen Is this disabling type check for the whole function body? A cleaner solution could be to use the `# type: ignore` notation on the offending lines if they are not too many.